semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -761,20 +767,21 @@ let lval_of_ent env ent =
       | [] -> raise Impossible
       | x :: _ -> fresh_lval env x)
 
-(* just to ensure the code after does not call expr directly *)
-let expr_orig = expr
+(* just to ensure the code after does not call expr directly
+ * TODO: Remove this if we can't figure out why it was done in the first place... *)
```

Comment:
I think the idea is that I wanted to prevent the code below to call expr directly; when you process stmts, you should
call expr_with_pre_stmts, not expr, which is why I kinda hide 'expr' by changing its type, so each use of
expr_orig below should be suspect.

Reply:
Aha! I have the feeling that I knew but I had forgotten, thanks for pointing it out. Now being `expr` and `stmt` mutually recursive it would be more cumbersome to ensure that, but I will add a comment at least.